Makefile

```diff
@@ -44,21 +44,15 @@ help: ## This help message
 show: ## show the starting template without installing it
 	helm template common/operator-install/ --name-template $(NAME) $(HELM_OPTS)
 
-# Only call helm install if the CRD is missing. If it already exists just
-# push the templated files.
-# The reason we have two helm template calls in the else branch is to avoid
-# warnings when the chart gets applied the first time, but the resources were
-# created first via the VP operator's UI
 .PHONY: operator-deploy
 operator-deploy operator-upgrade: validate-prereq validate-origin ## runs helm install
-	@set -e; if ! oc get crds patterns.gitops.hybrid-cloud-patterns.io >/dev/null 2>&1; then \
-	  echo "Running helm:"; \
-	  helm upgrade --install $(NAME) common/operator-install/ $(HELM_OPTS); \
-	else \
-	  echo "Reapplying helm chart:"; \
-	  helm template --name-template $(NAME) common/operator-install/ $(HELM_OPTS) | oc apply set-last-applied --create-annotation -f-; \
-	  helm template --name-template $(NAME) common/operator-install/ $(HELM_OPTS) | oc apply -f-; \
-	fi
+	@set -e -o pipefail
+	# We reapply the CRD and if it already exists we do not error out
+	oc apply common/operator-install/crds/*.yaml 2>/dev/null || /bin/true
+	# Retry five times because the CRD might not be fully installed yet
+	for i in {1..5}; do \
+		helm template --name-template $(NAME) common/operator-install/ $(HELM_OPTS) | oc apply -f- && break || sleep 10; \
+	done
```

Comment:
I think you want `helm template --include-crds` instead of this loop

Reply:
Nah you cannot avoid that loop :/ (believe me, I tried ;) ) . oc apply is not smart enough to wait for the CRD to be registered in the cluster. What I can do is drop the initial oc apply call for the CRD and just add --include-crds in the loop